viz_scripts/survey_metrics.ipynb

```diff
@@ -108,13 +108,16 @@
    "outputs": [],
    "source": [
     "#if fleet, replace primary_mode with primary_ble_sensed mode\n",
-    "if bluetooth_only and 'ble_sensed_summary' in expanded_ct_sensed.columns:\n",
-    "    expanded_ct_sensed = expanded_ct_sensed[expanded_ct_sensed['ble_sensed_summary'].notna()]\n",
-    "    expanded_ct_sensed[\"primary_mode\"] = expanded_ct_sensed.ble_sensed_summary.apply(lambda md: max(md[\"distance\"], key=md[\"distance\"].get))\n",
-    "    unique_keys = expanded_ct_sensed.groupby(\"primary_mode\").agg({distance_col: \"count\"}).index\n",
-    "    print(unique_keys)\n",
-    "    colors_mode, colors_purpose, colors_sensed = scaffolding.mapping_color_labels({},  dict(zip(unique_keys, unique_keys)), {})\n",
-    "    colors_sensed = colors_mode"
+    "try:\n",
+    "    if bluetooth_only and 'ble_sensed_summary' in expanded_ct_sensed.columns:\n",
+    "        expanded_ct_sensed = expanded_ct_sensed[expanded_ct_sensed['ble_sensed_summary'].notna()]\n",
+    "        expanded_ct_sensed[\"primary_mode\"] = expanded_ct_sensed.ble_sensed_summary.apply(lambda md: max(md[\"distance\"], key=md[\"distance\"].get))\n",
+    "        unique_keys = expanded_ct_sensed.groupby(\"primary_mode\").agg({distance_col: \"count\"}).index\n",
+    "        print(unique_keys)\n",
+    "        colors_mode, colors_replaced, colors_purpose, colors_sensed, colors_ble = await scaffolding.mapping_color_labels(unique_keys)\n",
+    "        colors_sensed = colors_ble\n",
+    "except ValueError as e:\n",
+    "    print(\"Got ValueError \", e)"
```

Comment:
What situation were you getting a value error in? If the deployment is bluetooth only and there are ble sensed summaries available, we want to replace the sensed modes with the ble modes.

Reply:
I wasn't getting any value error in this block of code. Rather, this block of code is identical to the one present in `generic_metrics_sensed` notebook which has try except block. So, I updated this code with try except block.

Reply:
`generic_metrics_sensed` generates the "average miles per sensed mode" chart, so rather than duplicate the chart I had chosen to update the chart in the original notebook. I could see an argument to move all of the `survey_metrics` to `generic_metrics_sensed` since the charts are all related to sensed modes, and then only display the `survey_metrics` charts on the dashboard for survey deployments (since they were designed to replace the `generic_metrics`.